kuma/attachments/management/commands/configure_github_social.py

```diff
@@ -0,0 +1,71 @@
+import fileinput, os, sys, urllib2
+
+from django.core.management.base import BaseCommand, CommandError
+from django.contrib.sites.models import Site
+from allauth.socialaccount.models import SocialApp
+
+LOCALHOST = 'localhost:8000'
+MDN_LOCALHOST = 'mdn.localhost'
+
+OVERWRITE_PROMPT = 'There\'s already a SocialApp for GitHub, if you want to overwrite it type "yes":'
+GITHUB_INFO = '\n'.join([
+    'Visit https://github.com/settings/developers and click "New OAuth App"',
+    'Set "Homepage URL" to "http://mdn.localhost:8000/" and Authorization callback URL to ' +
+    '"http://mdn.localhost:8000/users/github/login/callback/" respectively'
+])
+ENV_INFO = 'Putting SITE_ID and DOMAIN into .env'
+HOSTS_INFO = '\n'.join([
+    'Make sure your hosts file contains these lines:'
+    '127.0.0.1 kubernetes.docker.internal localhost demos mdn.localhost beta.mdn.localhost wiki.mdn.localhost',
+    '::1             mdn.localhost beta.mdn.localhost wiki.mdn.localhost'
+])
+
+
+def overwrite_or_create_env_vars(env_vars):
+    file_path = os.path.join(os.getcwd(), '.env')
+
+    for line in fileinput.input(file_path, inplace=True):
+        key = line.strip().split('=')[0]
+        if key not in env_vars:
+            sys.stdout.write(line)
+
+    with open(file_path, 'a') as file:
+        for key, value in env_vars.iteritems():
+            file.write('\n' + key + '=' + str(value))
+
+
+class Command(BaseCommand):
+    help = 'Configure Kuma for Sign-In with GitHub'
+
+    def handle(self, *args, **options):
+        print('\n')
+
+        social_app = SocialApp.objects.filter(provider='github').first()
+        if social_app is not None and raw_input(OVERWRITE_PROMPT) == 'yes':
+            print('\n')
+
+            print(GITHUB_INFO)
+            client_id = raw_input('Client ID: ')
+            client_secret = raw_input('Client Secret: ')
+
+            social_app, created = SocialApp.objects.update_or_create(
+                provider='github',
+                defaults={
+                    'name': 'MDN Development',
+                    'client_id': client_id,
+                    'secret': client_secret
+                }
+            )
+
+        site, created = Site.objects.update_or_create(
+            domain=LOCALHOST,
+            defaults={'name': LOCALHOST}
+        )
+        social_app.sites.add(site)
+
+        print('\n')
+
+        print(ENV_INFO)
+        overwrite_or_create_env_vars({'SITE_ID': site.id, 'DOMAIN': MDN_LOCALHOST})
```

Comment:
I don't know what's wrong with that function but it shouldn't have edited my file. 
```
▶ diff .env .env.backup
65a66
> DOMAIN=mdn.localhost
78a80
> DOMAIN=mdn.localhost
145,147d146
<
< DOMAIN=mdn.localhost
< SITE_ID=1
\ No newline at end of file
```

The `SITE_ID=site.id` should only happen if it's not already a key called `SITE_ID`. Because `SITE_ID = 1` is [already the default](https://github.com/mozilla/kuma/blob/6be5a95002bfe9e6750e36524421c9021a2ad50c/kuma/settings/common.py#L365).

Also, as you can see from by backup, I already had `DOMAIN=mdn.localhost` in mine yet it got added again.

Reply:
Also, when it does (correctly) add new entries the file fails to have 1 trailing linebreak.